internal/provider/resource_topic.go

```diff
@@ -46,17 +46,27 @@ func (t topicResourceType) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Di
 				Required:            true,
 				Type:                types.StringType,
 				Validators: []tfsdk.AttributeValidator{
-					validation.Compare(validation.OneOf, []string{"JSON", "Binary", "String", "Xml"}),
+					validation.Compare(validation.OneOf, []string{"AVRO", "JSON", "Binary", "String", "Xml"}),
 				},
 			},
+			"key_schema": {
```

Comment:
Please add validations here as well, let's call it "attribute level validations". I know you are doing the validations in code, "create level validations", but those validations only happen when doing "terraform apply". We would also like to have validations work during "terraform plan" because user of our provider would like to have certainty in their gitops pipeline that when tf plan is approved then tf apply will work without errors. 

In order to make attribute level validations work, here is sample pseudocode: https://pastebin.com/9r5NKpXi

The validation here should be: "if there is any value in key_schema, key_type has to be AVRO"